src/appmetrics.cpp

```diff
@@ -388,10 +397,22 @@ static void sendData(const std::string &sourceId, unsigned int size, void *data)
 	memcpy(dataCopy, data, size);
 	payload->data = dataCopy;
 	payload->size = size;
-
-	async->data = payload;
-	uv_async_init(uv_default_loop(), async, (uv_async_cb)emitMessage);
-	uv_async_send(async);
+	payload->next = NULL;
+
+	// Put the next message on the end of the queue.
+	// (So they are sent in the same order we added them.)
+	uv_mutex_lock(messageListMutex);
+	MessageData** tail = &messageList;
+	while( *tail != NULL ) {
+		tail = &((*tail)->next);
+	}
+	(*tail) = payload;
```

Comment:
We could, the list is just there to handle uv_async_send calls being coalesced.
If something was hogging the event loop the previous version of the code would have kept adding new uv_async_t's to the event loop. At least now we have visibility and can make that decision in our code if we need to. (I don't think a new problem has been introduced but you are right it might be an improvement we need in the future.)